tests/unit/test_is_stuck.py

```diff
@@ -314,8 +312,10 @@ def test_is_stuck_ipython_syntax_error_not_at_end(
         event_stream.add_event(ipython_observation_4, EventSource.USER)
 
         with patch('logging.Logger.warning') as mock_warning:
-            assert stuck_detector.is_stuck() is False
-            mock_warning.assert_not_called()
+            assert stuck_detector.is_stuck() is True
+            mock_warning.assert_called_once_with(
```

Comment:
Can we add the other errors in the tests? The tests here are way more useful, imho, than perhaps many others 😅: it's not trivial to read and assert naturally this kind of code, so the tests are really necessary.

Reply:
Yeh, I can add those, probably just by adding a loop of the list of errors, good point.